aopy/preproc/neuropixel.py

```diff
@@ -239,4 +240,41 @@ def sync_ts_data_timestamps(data, sync_timestamps):
         sync_data = data[not_crop_datapoints,:]
         sync_timestamps = sync_timestamps[not_crop_datapoints]
         
-    return np.squeeze(sync_data), sync_timestamps
\ No newline at end of file
+    return np.squeeze(sync_data), sync_timestamps
+
+def destripe_lfp_batch(lfp_data, save_path, sample_rate, bit_volts, batch_length = 180, dtype='int16'):
+    
+    '''
+    Destripe LFP data in each batch to save memory. The result is saved in save_path.
+    
+    Args:
+        lfp_data (nt, nch): LFP data. This should be a memory mapping array.
+        save_path (str): file path to save destriped lfp data
+        sample_rate (float): sampling rate in Hz
+        bit_volts (float): volt per bit
+        batch_length (int): batch size in second
+    
+    Returns:
+        None
+    
+    '''
+
+    # Load data and metadata
+    n_samples, _ = lfp_data.shape
+
+    # Create memmap array to save destriped lfp data
+    lfp_destriped = np.memmap(save_path, dtype=dtype, mode='w+', shape=lfp_data.shape)
+
+    # Destripe lfp
+    batch_size = int(sample_rate * batch_length)
+    Nbatches = np.ceil(n_samples/batch_size).astype(int)
+
+    for ibatch in range(Nbatches):
+        if ibatch < Nbatches-1:
+            tmp = destripe_lfp((lfp_data[ibatch*batch_size:(ibatch+1)*batch_size, :]*bit_volts).T/1e6, sample_rate)*1e6
+            lfp_destriped[ibatch*batch_size:(ibatch+1)*batch_size, :] = (tmp/bit_volts).T.astype(dtype)
+        else:
+            tmp = destripe_lfp((lfp_data[ibatch*batch_size:, :]*bit_volts).T/1e6, sample_rate)*1e6
+            lfp_destriped[ibatch*batch_size:, :] = (tmp/bit_volts).T.astype(dtype)

```

Comment:
I don't think this if/else statement is necessary. If you index into an array with a range that ends larger than the size of the array it should still work.